src/java.base/share/classes/sun/nio/ch/DatagramChannelImpl.java

```diff
@@ -1732,11 +1738,21 @@ private void implCloseBlockingMode() throws IOException {
                 long reader = readerThread;
                 long writer = writerThread;
                 if (reader != 0 || writer != 0) {
-                    nd.preClose(fd);
-                    if (reader != 0)
-                        NativeThread.signal(reader);
-                    if (writer != 0)
-                        NativeThread.signal(writer);
+                    boolean isAix = System.getProperty("os.name").startsWith("AIX");
```

Comment:
This needs to be concerned with the SecurityManager, so it should use `GetPropertyAction.privilegedGetProperty`. Also it shouldn't fetch "os.name" every time the method is called, but do it once and cache the result.

For an example see https://github.com/ibmruntimes/openj9-openjdk-jdk17/blob/openj9/src/java.base/share/classes/sun/security/provider/SunEntries.java#L98

Reply:
You could add a new public class sun.net.util.AIX and define a `public static final boolean isAix` variable there for every place to use.

Reply:
@pshipton Updated. Can you please take a look